### PR TITLE
RMQ: support for delayed message exchanges

### PR DESCRIPTION
**Pull request by rjgildea**
Delayed message exchanges specified with delayed_message_exchange
field in the rabbitmq configuration.

If no delayed message exchange is set, and a message is sent with
a delay, then raise a ValueError.
Otherwise send to the delayed message exchange if a delay is requested,
or to the default exchange if not.

### CODE REVIEW

**Review by Anthchirp — APPROVED** [review]
Fine I guess, but why not just assume a fixed name?

**Comment by rjgildea** [comment]
> Fine I guess, but why not just assume a fixed name?

Happy to assume a fixed name if that is preferable - it certainly makes this PR simpler. Any suggestions? `delayed`? `zocalo`?

File: src/workflows/transport/pika_transport.py [hunk 1]
```diff
@@ -505,10 +519,16 @@ def _send(
...
 
-        # if delay:
-        #     headers["x-delay"] = 1000 * delay
+        if delay:
+            if self._delayed_message_exchange is None:
+                raise ValueError(
+                    "No delayed message exchange set, but transport delay requested"
+                )
+            headers["x-delay"] = 1000 * delay
+            exchange = "zocalo"
```
Comment (rjgildea) [inline src/workflows/transport/pika_transport.py:529]:
```suggestion
            exchange = self._delayed_message_exchange
```

**Comment by ndevenish** [comment]
"delayed" seems a little clearer, because "zocalo" in a vhost named "zocalo" is about as clear as an exchange named "exchange", that you don't use for everything.

That said, since the exchange name is configurable and not implicitly created by the configuration, maybe having it explicitly specified in the configuration _is_ the right thing?

**Pull request merged**